common/src/main/java/dev/isxander/yacl3/api/controller/EnumDropdownControllerBuilder.java

```diff
@@ -0,0 +1,27 @@
+package dev.isxander.yacl3.api.controller;
+
+import dev.isxander.yacl3.api.Option;
+import dev.isxander.yacl3.impl.controller.EnumDropdownControllerBuilderImpl;
+
+import java.util.function.Function;
+
+public interface EnumDropdownControllerBuilder<E extends Enum<E>> extends ControllerBuilder<E> {
+    EnumDropdownControllerBuilder<E> toString(Function<E, String> toString);
+
+    static <E extends Enum<E>> EnumDropdownControllerBuilder<E> create(Option<E> option) {
+        return new EnumDropdownControllerBuilderImpl<>(option);
+    }
+
+    /**
+     * Creates a factory for {@link EnumDropdownControllerBuilder}s with the given function for converting enum constants to strings.
+     * Use this if a custom toString function for an enum is needed.
+     * Use it like this:
+     * <pre>{@code Option.<MyEnum>createBuilder().controller(createEnumDropdownControllerBuilder.getFactory(MY_CUSTOM_ENUM_TO_STRING_FUNCTION))}</pre>
+     * @param toString The function used to convert enum constants to strings used for display, suggestion, and validation
+     * @return a factory for {@link EnumDropdownControllerBuilder}s
+     * @param <E> the enum type
+     */
+    static <E extends Enum<E>> Function<Option<E>, ControllerBuilder<E>> getFactory(Function<E, String> toString) {
```

Comment:
A little confused as to what this toString function is? Should it not be a `Component`?

Reply:
This optional function is used to convert a enum to the string value displayed to the user. By default `Enum#toString` would be used. (Be default I mean not using this `getFactory` method and just using the create method above.) If there’s a better way to do this I can change this.

Reply:
You're supposed to use a value formatter for this. I believe there is an interface like ValueFormattableControllerBuilder or something.

Reply:
I took a look at value formatters, but they turn values (in this case enums) into components (or texts in yarn). However, I need a string for use with dropdowns (matching and parsing user input, etc), not just to display to the user. The only existing non-string dropdown controller, the item controller, uses `ResourceLocation#toString()`.

I don't see a better way to go about this.

Reply:
`Component#getString`. Be careful though, cache this result, it's not cheap.

Reply:
Also, shouldn't this be a builder method, not a static method?

Reply:
Because the components are rendered are they not? You get translations and formattings. If I let that through for item dropdowns that was a mistake and I'll look at changing that.